acceleration-everest-03-dotnet/DomainService/Services/CustomerService.cs

```diff
@@ -79,33 +90,35 @@ public void Update(Customer customerToUpdate)
             throw new ArgumentException($"Cpf: {customerToUpdate.Cpf} is already registered for Id: {existingId}");
         }
 
-        _context.Set<Customer>().Update(customerToUpdate);
-        _context.SaveChanges();
+        _unitOfWork.Repository<Customer>().Update(customerToUpdate);
+        _unitOfWork.SaveChanges();
     }
 
     private bool EmailAlreadyExists(Customer customerToCheck)
     {
-        return _context.Set<Customer>().Any(customer => customer.Email == customerToCheck.Email);
+        return _repositoryFactory.Repository<Customer>().Any(customer => customer.Email == customerToCheck.Email);
```

Comment:
Alocar esta implementação em um var.
```suggestion
        return .Any(customer => customer.Email == customerToCheck.Email);
```